src/fluree/db/query/fql/parse.cljc

```diff
@@ -653,15 +674,14 @@
         [vars values] (parse-values vals-map)
         where-map     {:where (util/get-first-value txn const/iri-where)}
         where         (parse-where where-map vars context)
-
         delete (-> (util/get-first-value txn const/iri-delete)
                    (json-ld/expand context)
                    (util/sequential)
-                   (parse-triples))
+                   (parse-triples where))
```

Comment:
I think it's simpler to collect a set of variables present in the where clause once here, and then pass that set to the `parse-triples` function as an `allowed-vars` argument. Then, instead of checking for variables in the where clause each time in `parse-triples` and downstream functions, we could and check each variable encountered in that triple for membership in the `allowed-vars` set. 

I think this method would minimize the number of times we had to traverse the where clause and we wouldn't have to worry about caches.

We also need to consider variables defined in the `:values` clause as well.